editor/src/components/canvas/remix/remix-utils.tsx

```diff
@@ -99,3 +128,405 @@ export function getRoutesFromRouteManifest(
     return []
   }
 }
+
+export function createAssetsManifest(routes: RouteManifest<EntryRoute>): AssetsManifest {
+  return {
+    entry: { imports: [], module: 'TODO' },
+    url: '/',
+    version: '1',
+    routes: routes,
+  }
+}
+
+export type RouteModuleWithFilePath = RouteModule & {
+  filePath: string
+}
+
+export interface RouteModulesWithFilePaths {
+  [routeId: string]: RouteModuleWithFilePath
+}
+
+export interface RouteModuleCreator {
+  filePath: string
+  executionScopeCreator: (projectContents: ProjectContentTreeRoot) => ExecutionScope
+}
+
+export interface RouteIdsToModuleCreators {
+  [routeId: string]: RouteModuleCreator
+}
+
+export interface RouteModulesWithRelativePaths {
+  [routeId: string]: {
+    relativePath: ElementPath
+    filePath: string
+  }
+}
+
+export interface GetRoutesAndModulesFromManifestResult {
+  routeModuleCreators: RouteIdsToModuleCreators
+  routes: Array<DataRouteObject>
+  routeModulesToRelativePaths: RouteModulesWithRelativePaths
+  routingTable: RemixStaticRoutingTable
+}
+
+export function getDefaultExportNameAndUidFromFile(
+  projectContents: ProjectContentTreeRoot,
+  filePath: string,
+): { name: string; uid: string } | null {
+  const file = getProjectFileByFilePath(projectContents, filePath)
+  if (file == null || file.type != 'TEXT_FILE' || file.lastParseSuccess == null) {
```

Comment:
Are we deliberately using `lastParseSuccess` here? It means that this function will return a result even if the current file is a `ParseFailure`, which I don't _think_ was the intention here?

Reply:
should it be `parsed`?

Reply:
`file.fileContents.parsed`

Reply:
https://github.com/concrete-utopia/utopia/pull/4162/commits/cddc103ffd7070810f4ed2a17a35d4afc97a1f4d